src/components/custom-button.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Spinner } from "./spinner";
+
+interface ButtonProps {
+  title: string;
+  onClick: () => void;
+  loading?: boolean;
+}
+
+export const CustomButton: React.FC<ButtonProps> = ({
+  title,
+  onClick,
+  loading,
+}) => {
```

Comment:
Opinião pessoal: eu acho que é mais válido extrair os valores na primeira linha da função doq ue quberar o protótipo dela. Porque por conta da identação, pode confundir um pouco na leitura:

```suggestion
export const CustomButton: React.FC<ButtonProps> = (props) => {
  const { title, loading, onClick } = props;
```